### ISSUE TITLE
why flip twice?

### ISSUE THREAD

**Issue by lxy5513** [soo89/CSD-SSD#14]
Thanks for you great work!
read the code, I find some  code which I can't to understand, as below,
<img width="680" alt="image" src="https://user-images.githubusercontent.com/39720528/117261081-503ba600-ae82-11eb-82a0-45111282f022.png">
why it need to flip  append loc and append conf?

---

by the way, we compare loc and flip loc in
```
consistency_loc_loss_x = torch.mean(torch.pow(loc_sampled[:, 0] + loc_sampled_flip[:, 0], exponent=2))
...
```
but, how we guarantee each item between loc_sampled and loc_sampled_flip can match correctly?

Hope your kindly response, much thanks.

**Comment by soo89** [comment]
Thank you for your interest.

 f^(p,r,c,d)_(**cls**)(I) should match with f^(p,r,C-c+1,d)_(**cls**)(I).
'L160 : append_conf = flip(append_conf,2)' is for matching above mentioned.

In our paper, we adopted not only classification loss but also regression loss. 

Therefore, we flipped loc outputs. 

 f^(p,r,c,d)_(**loc**)(I) should match with f^(p,r,C-c+1,d)_(**loc**)(I).
'L159 : append_loc = flip(append_loc,2)' is for matching above mentioned.
